src/routers/__tests__/SwitchRouter-test.js

```diff
@@ -200,10 +267,31 @@ describe('SwitchRouter', () => {
   });
 });
 
+// A simple helper that makes it easier to write basic routing tests
+// As we generally want to apply one action after the other,
+// it's often convenient to manipulate a structure that keeps the router state
+class ExampleRouterHelper {
```

Comment:
As a general rule I tend to avoid utilities or helpers in my tests, because they make it harder to see what is actually being tested. The rest of the tests in this library may be verbose, but they follow this philosophy.

For example, how often do the new tests check against the initial state? Only once. What if you pass a navigate action in with a null state? The helper makes it impossible to test that case.

That being said, this helper isn't the worst example, and the philosophy is just my opinion. So I'll leave it to Brent and your judgement on this one.

Reply:
I agree we should be consistent, but we could slowly migrate other tests if the best practice changes. @brentvatne do you have a preference?

Reply:
I agree with @slorber that it can be a bit tedious to make sure you're threading through the correct state variables. We should pull a class like this out to a helper file, call it something like `RouterTestHelper`, and have it accept a param for how it should initialize the router. Let's do that in a follow-up PR and change other tests to use it at the same time, and leave this PR with the helper as it is for now.

Reply:
ok, I'll try to do another PR. I'll do one for SwitchRouter first and then StackRouter. StackRouter has more complicated tests so it will mostly apply to tests of `getStateForAction`

Reply:
Sounds good!

(Also I can't help but humbly mention my preference for functions+return objects+closures over ES classes.. `const { applyAction, getState } = getRouterTestHelper(router);` )

Reply:
@ericvicenti - i'd be curious to see what that looks like next to the class version

Reply:
I can give a try in another PR and you compare. I'm fine with both as long as the tests are easy to read